### ISSUE TITLE
Feature Gamemaster Tool

### ISSUE THREAD

**Issue by Glenn-Lambert** [zen-mod/ZEN#305]
After a short talk to mharis001 id like to present an idea that might add value to ZEN.

A few years ago we developed a small gamemaster-tool in our clan, to accompany individual players and provide specific experiences to individual players that might increase the immersion of milsim. This tool was managed by and only available to zeus and the logged admin. It could be activated by a hotkey. 

**Basic Conditions of the gamemaster-tool:**
A. Per compact interface the zeus can trigger events on individual players.
B. Every activated event can be combined with free text as a hint that only the affected player can see, to explain what happened.
C. Combination of the events are possible

**Solution you'd like:**
We found events valuable such as:

1. Send normal hints to individual players

2. Send hintC to individual players, that must be acknowledged by the player with an "ok" button

3. Trip the player with a forced "lay down" animation, to simulate tripping in muddy environments.

4.  Force (on/off) walk with limping animation of the player to simulate injuries, independently from vanilla or ace injuries.

5. Toggle unconsciousness (in combination with the free descriptive hint)

6. Toggle inability to act, player lays on the ground, similar to unconsciousness, but can see and move his head.

7. Time-limited and permanent visual effects on specific players, such as:
a. Freezing (camera shake with light blue vision),
b. overheat/ heat squatting (bleaching of vision)
c. poisoning by snakebites (frequently and slightly changing colours)
d. different types of pain
e. fatigue/tiredness (blinking eyes effect)
f. drug intoxication (alcohol, cocain, mushrooms) (we can provide different setups of colour visions)

8. Drop the players gun to the ground. (Because of weakness, accidents or smth.)

9. Remove the magazine of the players current weapon to simulate combat-damage on a magazin.

10. Jamming, permanent malfuncion or exploding gun in players hands. If a player uses a combat-damaged gun or a cheap chinese imitate, the zeus can be able to trigger an harmful event for the player. Since guns in arma do not have id's, there was no reference to the gun itselfe, but only to the player.

**Alternatives you've considered:**
We used a compact interface with a list of all active players, which was very useful to speed up managing process for the gamemaster. Other types of activations are imaginable, but not useful as separate functions. The combination of the explanatory hint with the event is the key to the players immersive experience.

**Additional context:**
Link of the simple interface:
![Screenshot_ares_gamemaster_1](https://user-images.githubusercontent.com/39874155/74877790-540b0780-5366-11ea-9f5c-8be229a5b435.jpg)

Due to loss of our competent magic-scripter, we are not able anymore to develop and repair our game-master tool. Therefore we are looking for a successful project to hand over our idea.
If there are questions or anything you require, please let me know. I hope this idea will find a new home in ZEN.

Best greetings - Lambert from ARES Military

**Comment by CreepPork** [comment]
I think this is overly specific and ZEN should be more about giving users good standards and frameworks rather than giving a clan-specific module.

I'd still agree with some points, such as 1., 2. and partially the 7th point.

The medical status (limping, unconsciousness) should only be reflected because of the player's actual injuries and not the arbitrary will of the Curator.

Jamming, weapon malfunctions etc. isn't, in my opinion, an issue that should be addressed in ZEN but rather something like ACE or a separate mod, it doesn't really fit into Zeus as such.

In conclusion, some ideas are good but there are way too specific things included in this feature request that just doesn't seem should be a part of ZEN, for example, snakebite effects.

Perhaps @mharis001 could specify his intentions for this issue and how it would fit into ZEN without being side-tracked with too specific things.

**Comment by mharis001** [comment]
@Glenn-Lambert reached out in terms of some ideas from his community's personal mods possibly being integrated into ZEN. I suggested he make a feature request on GitHub to outline what these might be. After reading this, I agree with @CreepPork - this likely too specific (for lack of a better word) for inclusion into ZEN.

**Comment by Glenn-Lambert** [comment]
Maybe there are Ideas that you might find useful. If not, thats ok. Then i will try to find another solution.
